### PR TITLE
Update to Community Hub content

### PR DESCRIPTION
**Pull request by suuus**
## Hey, I just made a Pull Request!

I've made the following changes for the Community Hub:
- Added past recordings of Community Sessions
- Changed sort mechanism for on-demand
- Updated documentation to reflect the changes
- Improved language a bit for the /community page

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: microsite/data/on-demand/20211118-01.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+title: Community Sessions
+date: November 18, 2021
+category: Meetup
+description: In the contributors track for this month’s community session, hear about upcoming deprecations, the fruits of Hacktoberfest, the future of both the Kubernetes plugin and the Tech Insights plugin, tips on contributing, and more. .
```
Comment (freben) [inline microsite/data/on-demand/20211118-01.yaml:5]: extra dot at end

File: microsite/data/on-demand/20220323.yaml [hunk 2]
```diff
@@ -4,6 +4,6 @@ date: March 23, 2022
 category: Upcoming
 description: Join the maintainers and contributors for the Contributor Community Sessions
 youtubeUrl: https://youtu.be/evf_LV0KzIk
-youtubeImgUrl: https://i1.ytimg.com/vi/evf_LV0KzIk/mqdefault.jpg
+youtubeImgUrl: http://localhost:3000/img/b-sessions.png
```
Comment (freben) [inline microsite/data/on-demand/20220323.yaml:7]: oops

File: microsite/pages/en/on-demand.js [hunk 3]
```diff
@@ -17,7 +17,7 @@ const ondemandDirectory = require('path').join(process.cwd(), 'data/on-demand');
 const ondemandMetadata = fs
   .readdirSync(ondemandDirectory)
   .map(file => yaml.load(fs.readFileSync(`./data/on-demand/${file}`, 'utf8')))
-  .sort((b, a) => a.date.toLowerCase().localeCompare(b.date.toLowerCase()));
+  .reverse();
```
Comment (freben) [inline microsite/pages/en/on-demand.js:20]: i'm not so sure that it's actually guaranteed that files are read in any sensible or even deterministic order - you probably need to add a `sort` before the `map`

File: microsite/data/on-demand/20211117-01.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+---
+title: Community Sessions
+date: November 17, 2021
+category: Meetup
+description: At this month’s adopters session, get the scoop on Spotify’s plan for paid plugins, a first look at Box’s DevPortal, and answers to big questions, like “How will I know if Backstage will work at a large company like mine?”
```
Comment (freben) [inline microsite/data/on-demand/20211117-01.yaml:5]: Does your locale have some odd quotation mark setting? :) I know it's a huge nitpick but it caught my eye that they looked odd

File: microsite/pages/en/on-demand.js (hunk 3 above)
Comment (suuus) [reply microsite/pages/en/on-demand.js:20]: readdirSync follows the systems default sort, the filenames are made up of the dates in yyyymmdd. I reverse the whole thing to make sure the latest ones show up first. Happy to implement some other suggestion, I chose this approach to keep the date field in the file itself human friendly.

Comment (freben) [reply microsite/pages/en/on-demand.js:20]: I just don't trust what the default sort is :) especially across operating systems. Just moving the sort to be on the file names before reading their contents would make it guaranteed to be deterministic and also declares intent nicely.

**Review by freben — APPROVED** [review]

**Pull request merged**